boost host build fails on Gentoo

Maintainer: @ClaymorePT 
Environment: OpenWrt master r10127-3209f5ae3d

Description: the host build of boost does not compile on Gentoo

```
make[2]: Entering directory '/home/stijn/Development/LEDE/openwrt-packages/libs/boost'
Selected Boost API sysv for architecture x86_64 and cpu type " "
. /home/stijn/Development/LEDE/source/include/shell.sh; bzcat /home/stijn/Development/LEDE/source/dl/boost_1_70_0.tar.bz2 | tar -C /home/stijn/Development/LEDE/source/build_dir/hostpkg/boost_1_70_0/.. -xf -
[ ! -d ./src/ ] || cp -fpR ./src/* /home/stijn/Development/LEDE/source/build_dir/hostpkg/boost_1_70_0

Applying ./patches/010-uclibc-ng.patch using plaintext:
patching file boost/asio/detail/impl/eventfd_select_interrupter.ipp

Applying ./patches/100-fix-apply_visitor-on-lvalue-variant.patch using plaintext:
patching file boost/variant/detail/apply_visitor_unary.hpp
patching file libs/variant/test/const_ref_apply_visitor.cpp
Hunk #1 succeeded at 217 (offset -7 lines).
Hunk #2 succeeded at 371 (offset -11 lines).

Applying ./patches/uclibc-ng-2.patch using plaintext:
patching file boost/math/tools/roots.hpp
touch /home/stijn/Development/LEDE/source/build_dir/hostpkg/boost_1_70_0/.prepared3414f67d9cf4e433bcc595300cd62562_6664517399ebbbc92a37c5bb081b5c53
(cd /home/stijn/Development/LEDE/source/build_dir/hostpkg/boost_1_70_0/; if [ -x configure ]; then cp -fpR /home/stijn/Development/LEDE/source/scripts/config.{guess,sub} /home/stijn/Development/LEDE/source/build_dir/hostpkg/boost_1_70_0// && CC="gcc" CFLAGS="-O2 -I/home/stijn/Development/LEDE/source/staging_dir/host/include -I/home/stijn/Development/LEDE/source/staging_dir/hostpkg/include -I/home/stijn/Development/LEDE/source/staging_dir/target-x86_64_musl/host/include" CXX="g++" CPPFLAGS="-I/home/stijn/Development/LEDE/source/staging_dir/host/include -I/home/stijn/Development/LEDE/source/staging_dir/hostpkg/include -I/home/stijn/Development/LEDE/source/staging_dir/target-x86_64_musl/host/include" LDFLAGS="-L/home/stijn/Development/LEDE/source/staging_dir/host/lib -L/home/stijn/Development/LEDE/source/staging_dir/hostpkg/lib -L/home/stijn/Development/LEDE/source/staging_dir/target-x86_64_musl/host/lib" CONFIG_SHELL="/usr/bin/env bash"  bash ./configure --target=x86_64-pc-linux-gnu --host=x86_64-pc-linux-gnu --build=x86_64-pc-linux-gnu --program-prefix="" --program-suffix="" --prefix=/home/stijn/Development/LEDE/source/staging_dir/hostpkg --exec-prefix=/home/stijn/Development/LEDE/source/staging_dir/hostpkg --sysconfdir=/home/stijn/Development/LEDE/source/staging_dir/hostpkg/etc --localstatedir=/home/stijn/Development/LEDE/source/staging_dir/hostpkg/var --sbindir=/home/stijn/Development/LEDE/source/staging_dir/hostpkg/bin ; fi )
touch /home/stijn/Development/LEDE/source/build_dir/hostpkg/boost_1_70_0/.configured
# b2 does not provide a configure-script nor a Makefile
( cd /home/stijn/Development/LEDE/source/build_dir/hostpkg/boost_1_70_0/tools/build/src/engine ; ./build.sh gcc )
###
### Using 'gcc' toolset.
###
rm -rf bootstrap
mkdir bootstrap
gcc -o bootstrap/jam0 command.c compile.c constants.c debug.c execcmd.c frames.c function.c glob.c hash.c hdrmacro.c headers.c jam.c jambase.c jamgram.c lists.c make.c make1.c object.c option.c output.c parse.c pathsys.c regexp.c rules.c scan.c search.c subst.c timestamp.c variable.c modules.c strings.c filesys.c builtins.c class.c cwd.c native.c md5.c w32_getreg.c modules/set.c modules/path.c modules/regex.c modules/property-set.c modules/sequence.c modules/order.c execunix.c fileunix.c pathunix.c
./bootstrap/jam0 -f build.jam --toolset=gcc --toolset-root= clean
...found 1 target...
...updating 1 target...
...updated 1 target...
./bootstrap/jam0 -f build.jam --toolset=gcc --toolset-root=
...found 160 targets...
...updating 3 targets...
[MKDIR] bin.linuxx86_64
[COMPILE] bin.linuxx86_64/b2
debugger.c: In function 'debug_list_read':
debugger.c:165:5: warning: ignoring return value of 'fscanf', declared with attribute warn_unused_result [-Wunused-result]
     fscanf( in, "%d", &len );
     ^~~~~~~~~~~~~~~~~~~~~~~~
[COPY] bin.linuxx86_64/bjam
...updated 3 targets...
( cd /home/stijn/Development/LEDE/source/build_dir/hostpkg/boost_1_70_0 ; ./bootstrap.sh --prefix=/home/stijn/Development/LEDE/source/staging_dir/host --with-libraries=headers ; ./b2 install )
Building Boost.Build engine with toolset gcc... tools/build/src/engine/bin.linuxx86_64/b2
Unicode/ICU support for Boost.Regex?... /usr
Generating Boost.Build configuration in project-config.jam for gcc...

Bootstrapping is done. To build, run:

    ./b2

To generate header files, run:

    ./b2 headers

To adjust configuration, edit 'project-config.jam'.
Further information:

   - Command line help:
     ./b2 --help

   - Getting started guide:
     http://www.boost.org/more/getting_started/unix-variants.html

   - Boost.Build documentation:
     http://www.boost.org/build/

/home/build/openwrt/build_dir/hostpkg/boost_1_70_0/tools/build/src/build/feature.jam:491: in feature.validate-value-string from module feature
error: "none" is not a known value of feature <optimization>
error: legal values: "off" "speed" "space"
/home/build/openwrt/build_dir/hostpkg/boost_1_70_0/tools/build/src/build/property.jam:341: in validate1 from module property
/home/build/openwrt/build_dir/hostpkg/boost_1_70_0/tools/build/src/build/property.jam:367: in property.validate from module property
/home/build/openwrt/build_dir/hostpkg/boost_1_70_0/tools/build/src/tools/features/variant-feature.jam:70: in variant from module variant-feature
/usr/share/boost-build/site-config.jam:9: in modules.load from module site-config
/home/build/openwrt/build_dir/hostpkg/boost_1_70_0/tools/build/src/build-system.jam:255: in load-config from module build-system
/home/build/openwrt/build_dir/hostpkg/boost_1_70_0/tools/build/src/build-system.jam:425: in load-configuration-files from module build-system
/home/build/openwrt/build_dir/hostpkg/boost_1_70_0/tools/build/src/build-system.jam:607: in load from module build-system
/home/build/openwrt/build_dir/hostpkg/boost_1_70_0/tools/build/src/kernel/modules.jam:295: in import from module modules
/home/build/openwrt/build_dir/hostpkg/boost_1_70_0/tools/build/src/kernel/bootstrap.jam:139: in boost-build from module
/home/build/openwrt/build_dir/hostpkg/boost_1_70_0/boost-build.jam:17: in module scope from module

make[2]: *** [Makefile:555: /home/stijn/Development/LEDE/source/build_dir/hostpkg/boost_1_70_0/.built] Error 1
make[2]: Leaving directory '/home/stijn/Development/LEDE/openwrt-packages/libs/boost'
time: package/feeds/packages/boost/host-compile#34.41#9.97#44.00
make[1]: *** [package/Makefile:107: package/feeds/packages/boost/host/compile] Error 2
make[1]: Leaving directory '/home/stijn/Development/LEDE/source'
make: *** [/home/stijn/Development/LEDE/source/include/toplevel.mk:218: package/boost/compile] Error 2
```

Reverting 7c42ddf9ff6215104a749b0b5d271f764f6cf4c7 makes the problem go away.

--- a/libs/boost/Makefile
+++ b/libs/boost/Makefile
@@ -384,7 +384,7 @@ define Host/Compile
 				headers \
 				$(foreach lib,$(BOOST_LIBS), \
 					$(if $(findstring python,$(lib)),,$(if $(CONFIG_boost-host-build-$(lib)),$(lib)))))) ; \
-		./b2 install )
+		./b2 --ignore-site-config install )
 endef
 
 CONFIGURE_PREFIX:=$(PKG_INSTALL_DIR)

--- a/libs/boost/patches/010-uclibc-ng.patch
+++ b/libs/boost/patches/010-uclibc-ng.patch
@@ -15,7 +15,7 @@
  #include <boost/asio/detail/cstdint.hpp>
  #include <boost/asio/detail/eventfd_select_interrupter.hpp>
  #include <boost/asio/detail/throw_error.hpp>
-@@ -46,14 +46,14 @@ eventfd_select_interrupter::eventfd_select_interrupter()
+@@ -46,14 +46,14 @@ eventfd_select_interrupter::eventfd_sele
  
  void eventfd_select_interrupter::open_descriptors()
  {
@@ -32,7 +32,7 @@
  # if defined(EFD_CLOEXEC) && defined(EFD_NONBLOCK)
    write_descriptor_ = read_descriptor_ =
      ::eventfd(0, EFD_CLOEXEC | EFD_NONBLOCK);
-@@ -70,7 +70,7 @@ void eventfd_select_interrupter::open_descriptors()
+@@ -70,7 +70,7 @@ void eventfd_select_interrupter::open_de
        ::fcntl(read_descriptor_, F_SETFD, FD_CLOEXEC);
      }
    }

--- a/libs/boost/patches/100-fix-apply_visitor-on-lvalue-variant.patch
+++ b/libs/boost/patches/100-fix-apply_visitor-on-lvalue-variant.patch
@@ -8,8 +8,6 @@ Ensures that we can e.g. call apply_visitor on a lvalue variant
  boost/variant/detail/apply_visitor_unary.hpp | 9 +++++----
  1 file changed, 5 insertions(+), 4 deletions(-)
 
-diff --git a/boost/variant/detail/apply_visitor_unary.hpp b/boost/variant/detail/apply_visitor_unary.hpp
-index 71610cb6..c3a741ef 100644
 --- a/boost/variant/detail/apply_visitor_unary.hpp
 +++ b/boost/variant/detail/apply_visitor_unary.hpp
 @@ -23,6 +23,7 @@
@@ -46,7 +44,7 @@ index 71610cb6..c3a741ef 100644
      };
  
      typedef typename deduce_impl<begin_it>::type type;
-@@ -132,7 +133,7 @@ inline decltype(auto) apply_visitor(Visitor&& visitor, Visitable&& visitable,
+@@ -132,7 +133,7 @@ inline decltype(auto) apply_visitor(Visi
          boost::detail::variant::has_result_type<Visitor>
      >::type* = 0)
  {
@@ -55,21 +53,9 @@ index 71610cb6..c3a741ef 100644
      return ::boost::forward<Visitable>(visitable).apply_visitor(cpp14_vis);
  }
  
-
-From 7331d648f46b6008138cbd22087b8c5c61ff7926 Mon Sep 17 00:00:00 2001
-From: Ed Catmur <ed@catmur.uk>
-Date: Sun, 21 Apr 2019 00:50:00 +0100
-Subject: [PATCH 2/2] Add test.
-
----
- libs/variant/test/const_ref_apply_visitor.cpp | 44 ++++++++++++++++++++++++++++++++
- 1 file changed, 44 insertions(+)
-
-diff --git a/libs/variant/test/const_ref_apply_visitor.cpp b/libs/variant/test/const_ref_apply_visitor.cpp
-index 0ec77bd4..72afff9e 100644
 --- a/libs/variant/test/const_ref_apply_visitor.cpp
 +++ b/libs/variant/test/const_ref_apply_visitor.cpp
-@@ -224,6 +224,44 @@ void test_cpp14_visitor(const variant_type& test_var, const variant_type& test_v
+@@ -217,6 +217,44 @@ void test_cpp14_visitor(const variant_ty
  #endif
  }
  
@@ -114,7 +100,7 @@ index 0ec77bd4..72afff9e 100644
  void test_cpp14_visitor(variant_type&& test_var)
  {
      std::cout << "Testing rvalue visitable for c++14\n";
-@@ -344,8 +382,14 @@ void run_cpp14_mixed_tests()
+@@ -333,8 +371,14 @@ void run_cpp14_mixed_tests()
  void run_cpp14_tests()
  {
  #ifndef BOOST_NO_CXX14_DECLTYPE_AUTO

